src/Controller/Admin/TournamentController.php

```diff
@@ -156,4 +171,35 @@ private function canGenerateRound(Tournament $tournament): bool
                 && $tournament->getStatus() === Tournament::STATUS_PLANNED
             );
     }
+
+    public function setSwissTournamentResultsAction(Request $request, SwissTournamentManager $swissTournamentManager, $tournamentId)
+    {
+        $roundResults = $this->getDoctrine()->getRepository(RoundResult::class)->findBy(['tournament' => $tournamentId]);
+
+        $participants = $this->swissTournamentManager->getParticipantsDataByTournamentId($tournamentId);
+
+        return $this->render('admin/save_results.html.twig', [
+            'round_results' => $roundResults,
+            'participants' => $participants,
+            'tournament_id' => $tournamentId,
+        ]);
+    }
+
+    public function setSwissRoundResultsAction($tournamentId, $roundId)
+    {
+        $roundResult = $this->getDoctrine()->getRepository(RoundResult::class)->find($roundId);
+
+        $blackParticipantId = $roundResult->getBlackParticipant()->getId();
+        $whiteParticipantId = $roundResult->getWhiteParticipant()->getId();
+
+        $blackParticipant = $this->getDoctrine()->getRepository(Participant::class)->find($blackParticipantId);
```

Comment:
Don't retreive the same object twice from the database. $roundResult->getBlackParticipant() - that is the guy you are looking for, no need for extra queries